src/utils/CustomPropTypes/CustomPropTypes.js

```diff
@@ -0,0 +1,35 @@
+
+const range = (start, end) => {
+    return (props, propName, componentName) => {
+        const value = props[propName];
+
+        // If no value is provided, don't fail validation.
+        if (typeof value === 'undefined' || value === null) {
+            return null;
+        }
+
+        if (isNaN(value)) {
+            return new TypeError(`${componentName}: ${propName} must be a non-NaN number.`);
+        }
+
+        if (isNaN(start)) {
+            return new TypeError(`${componentName}: Range start must be a non-NaN number.`);
+        }
+
+        if (isNaN(end)) {
+            return new TypeError(`${componentName}: Range end must be a non-NaN number.`);
+        }
+
+        if (start > end) {
+            return new Error(`${componentName}: Range start must be less than or equal to range end.`);
+        }
+
+        if (value >= start && value <= end) {
+            return null;
+        } else {
+            return new Error(`${componentName}: ${propName} must be within the range ${start} to ${end}`);
+        }
+    };
+};
+
+export default {range};
```

Comment:
I'll just include some comments for this file without targeting each individual line.

1. Would `start` and `end` be better as `minVal` and `maxVal`?
1. Any validation that occurs on the `start` and `end` parameter values must be in the outer function -- basically meaning that we throw the error (or console log it) and then don't return the validation function.
1. Personally, I would probably just throw `Error` for everything rather than using `TypeError`.  I guess it technically doesn't matter, but not quite sure what you get with `TypeError` that `Error` doesn't give you.
1. The `isNaN` checks are correct, but I'd reword the error messages.  Error messages should be easy-to-read English sentences (see examples below).
1. The convention within `PropTypes` for error messages is to put relevant data (prop names, component names, data types, values, etc.) in backticks so they stand out in the message.

Below are a couple example prop validation errors coming from `PropTypes`:
```
The prop `className` is marked as required in `ActionBar`, but its value is `undefined`.

Invalid prop `glyph` of type `number` supplied to `Identifier`, expected `string`.
```
Maybe these should be something like this:
```
Invalid prop `color` of type `string` supplied to `Identifier`, expected `number`.

The prop `color` supplied to `Identifier` has a value of `10`, which is not in the range `1` to `9` (inclusive).
```

Reply:
I was following https://github.com/airbnb/prop-types/blob/master/src/between.js as an example. I think TypeError is more explicit about why the validation is failing. But I do agree that we can make the messages easier to read and more descriptive.

Reply:
The Airbnb one is a custom extension library whereas `prop-types` (from Facebook) is the defacto standard.  I'd rather pattern after `prop-types` so all prop type validation messages look similar.